src/Compilers/Core/Portable/Diagnostic/SuppressionDescriptor.cs

```diff
@@ -0,0 +1,106 @@
+        /// </summary>
+        public string SuppressedDiagnosticId { get; }
+
+        /// <summary>
+        /// A localizable justification about the suppression.
+        /// </summary>
+        public LocalizableString Justification { get; }
+
+        /// <summary>
+        /// Create a SuppressionDescriptor, which provides a justification about a programmatic suppression of a <see cref="Diagnostic"/>.
+        /// NOTE: For localizable <paramref name="justification"/>,
+        /// use constructor overload <see cref="SuppressionDescriptor(string, string, LocalizableString)"/>.
+        /// </summary>
+        /// <param name="id">A unique identifier for the suppression. For example, suppression ID "SP1001".</param>
+        /// <param name="suppressedDiagnosticId">Identifier of the suppressed diagnostic, i.e. <see cref="Diagnostic.Id"/>. For example, compiler warning Id "CS0649".</param>
+        /// <param name="justification">Justification for the suppression. For example: "Suppress CS0649 on fields marked with YYY attribute as they are implicitly assigned.".</param>
+        public SuppressionDescriptor(
+            string id,
+            string suppressedDiagnosticId,
+            string justification)
+            : this(id, suppressedDiagnosticId, (LocalizableString)justification)
+        {
+        }
+
+        /// <summary>
+        /// Create a SuppressionDescriptor, which provides a localizable justification about a programmatic suppression of a <see cref="Diagnostic"/>.
+        /// </summary>
+        /// <param name="id">A unique identifier for the suppression. For example, suppression ID "SP1001".</param>
+        /// <param name="suppressedDiagnosticId">Identifier of the suppressed diagnostic, i.e. <see cref="Diagnostic.Id"/>. For example, compiler warning Id "CS0649".</param>
+        /// <param name="justification">Justification for the suppression. For example: "Suppress CS0649 on fields marked with YYY attribute as they are implicitly assigned.".</param>
+        public SuppressionDescriptor(
+            string id,
+            string suppressedDiagnosticId,
+            LocalizableString justification)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+            {
+                throw new ArgumentException(CodeAnalysisResources.SuppressionIdCantBeNullOrWhitespace, nameof(id));
+            }
+
+            if (string.IsNullOrWhiteSpace(suppressedDiagnosticId))
+            {
+                throw new ArgumentException(CodeAnalysisResources.DiagnosticIdCantBeNullOrWhitespace, nameof(suppressedDiagnosticId));
+            }
+
+            this.Id = id;
+            this.SuppressedDiagnosticId = suppressedDiagnosticId;
+            this.Justification = justification ?? throw new ArgumentNullException(nameof(justification));
+        }
+
+        public bool Equals(SuppressionDescriptor other)
+        {
+            return
```

Comment:
>return [](start = 12, length = 6)

Consider checking if this is the same instance before doing more expensive comparison.